libcloud/compute/drivers/kubevirt.py

```diff
@@ -0,0 +1,996 @@
+            '/virtualmachines/' + name
+        data = {"spec": {"running": True}}
+        headers = {"Content-Type": "application/merge-patch+json"}
+        try:
+            result = self.connection.request(req, method="PATCH",
+                                             data=json.dumps(data),
+                                             headers=headers)
+
+            return result.status in VALID_RESPONSE_CODES
+
+        except Exception as exc:
+            raise
+
+    def ex_stop_node(self, node):
+        # check if running
+        if node.state is NodeState.STOPPED:
+            return True
+        name = node.name
+        namespace = node.extra['namespace']
+        req = KUBEVIRT_URL + 'namespaces/' + namespace + \
+            '/virtualmachines/' + name
+        headers = {"Content-Type": "application/merge-patch+json"}
+        data = {"spec": {"running": False}}
+        try:
+            result = self.connection.request(req, method="PATCH",
+                                             data=json.dumps(data),
+                                             headers=headers)
+
+            return result.status in VALID_RESPONSE_CODES
+
+        except Exception as exc:
+            raise
+
+    def reboot_node(self, node):
+        """
+        Rebooting a node.
+        """
+        namespace = node.extra['namespace']
+        name = node.name
+        method = 'DELETE'
+        try:
+            result = self.connection.request(KUBEVIRT_URL + 'namespaces/' +
+                                             namespace +
+                                             '/virtualmachineinstances/' +
+                                             name,
+                                             method=method)
+
+            return result.status in VALID_RESPONSE_CODES
+        except Exception as e:
+            raise
+        return
+
+    def destroy_node(self, node):
+        """
+        Terminating a VMI and deleting the VM resource backing it
+        """
+        namespace = node.extra['namespace']
+        name = node.name
+        # stop the vmi first
+        self.ex_stop_node(node)
+
+        try:
+            result = self.connection.request(KUBEVIRT_URL + 'namespaces/' +
+                                             namespace +
+                                             '/virtualmachines/' + name,
+                                             method='DELETE')
+            return result.status in VALID_RESPONSE_CODES
+        except Exception as exc:
+            raise
+
+    # only has container disk support atm with no persistency
+    def create_node(self, **kwargs):
+        """
+        Creating a VM with a containerDisk.
+        :param name: A name to give the VM. The VM will be identified by
+                     this name and atm it cannot be changed after it is set.
+        :type name: ``str``
+
+        :param namespace: The namespace where the VM will live.
```

Comment:
All the arguments which are not part of the standard API should be prefixed with ``ex_`` (namespace, memory, cpu, disks, network).